Makefile

```diff
@@ -111,12 +111,16 @@ API_GO_FILES = api/v1alpha1/zz_cache.pb.go api/v1alpha1/zz_rules.pb.go
 manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
 	$(CONTROLLER_GEN) rbac:roleName=manager-role crd:ignoreUnexportedFields=true webhook paths="./..." output:crd:artifacts:config=config/crd/bases
 
-.PHONY: update-git-submodules
-update-git-submodules: ## Pull the latest changes from the API main branch
+.PHONY: update-git-submodules-remote
+update-git-submodules-remote: ## Pull the latest changes from remote upstream HEAD
 	git submodule update --init --remote
 
+.PHONY: update-git-submodules
+update-git-submodules: ## Sync submodule content with parent project checkout
+	git submodule update --init
+
 .PHONY: gingersnap-api-generate
-gingersnap-api-generate: protoc protoc-gen-go protoc-gen-deepcopy $(API_GO_FILES) ## Generate code for gingersnap-api
+gingersnap-api-generate: update-git-submodules protoc protoc-gen-go protoc-gen-deepcopy $(API_GO_FILES) ## Generate code for gingersnap-api
```

Comment:
IMO we shouldn't add `update-git-submodules`  to the generate target. What if I want to pull from remote then generate?

If we omit `update-git-submodules` on `gingersnap-api-generate` we can do either:

`make update-git-submodules-remote  generate` or `make update-git-submodules generate`

Reply:
`update-git-submodules` only ensure that the content of gingersnap-api folder is aligned with the commit "pointed" by the parent project. I think it's safe to check it before each build.

`update-git-submodules-remote` does something different: it updates the parent project to point to the HEAD of the submodule branch and then updates the gingersnap-api folder.
So basically if you run a remote update, the following executions of `update-git-submodules` just ensure that the folder is aligned with the remote HEAD.

well let me check if all this is true :)

Reply:
Thanks for confirming